p2p/libp2p/defineOptions_test.go

```diff
@@ -0,0 +1,133 @@
+package libp2p
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/mock"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/stretchr/testify/assert"
+)
+
+func createStubMessengerForDefineOptions(notifeeCalled func(), setStreamHandlerCalled func()) *networkMessenger {
+	stubHost := &mock.ConnectableHostStub{
+		EventBusCalled:  nil,
+		IDCalled:        nil,
+		PeerstoreCalled: nil,
+		AddrsCalled:     nil,
+		NetworkCalled: func() network.Network {
+			return &mock.NetworkStub{
+				NotifyCalled: func(notifiee network.Notifiee) {
+					notifeeCalled()
+				},
+			}
+		},
+		MuxCalled:     nil,
+		ConnectCalled: nil,
+		SetStreamHandlerCalled: func(pid protocol.ID, handler network.StreamHandler) {
+			setStreamHandlerCalled()
+		},
+		SetStreamHandlerMatchCalled: nil,
+		RemoveStreamHandlerCalled:   nil,
+		NewStreamCalled:             nil,
+		CloseCalled:                 nil,
+		ConnManagerCalled:           nil,
+		ConnectToPeerCalled:         nil,
+	}
+
+	mes := &networkMessenger{
+		ctxProvider: &Libp2pContext{
+			ctx:      context.Background(),
+			connHost: stubHost,
+		},
+	}
+
+	return mes
+}
+
+func createStubMessengerFailingIfTriggered(t *testing.T) *networkMessenger {
+	return createStubMessengerForDefineOptions(
+		func() {
```

Comment:
hard to understand what does effectively the func represent. Add the name of the function parameter.

Reply:
added variables for the anonymous functions